src/libraries/Microsoft.Extensions.Options/tests/SourceGeneration.Unit.Tests/Main.cs

```diff
@@ -1264,6 +1266,123 @@ public class MyOptions
         File.Delete(assemblyPath); // cleanup
     }
 
+    [ConditionalFact(nameof(SupportRemoteExecutionAndNotInBrowser))]
+    public void IgnoreNotAccessibleValidationAttributesTest()
+    {
+        string source = """
+            using System;
+            using System.ComponentModel.DataAnnotations;
+
+            #nullable enable
+
+            namespace ValidationTest;
+
+            public class BaseOptions
+            {
+                [Timeout] // internal attribute not visible outside the assembly
+                public int Prop1 { get; set; }
```

Comment:
What if `TimeoutAttribute` is `public`, but the `Prop1` is `internal` - would we validate it?

Reply:
The change here is not changing this logic because we check the accessibility of the validation attribute. But this is an interesting other case I can try to see how the generator behaves here.

Reply:
@xakep139 I checked this behavior. The validator is enumerating the public properties only from the base class. So never validate internal properties from the base classes.

Reply:
Could you also add a few tests on private attributes? i.e.
```C#
public class BaseOptions
{
    [Timeout] // internal attribute not visible outside the assembly
    public int Prop1 { get; set; }

    private class TimeoutAttribute : ValidationAttribute { }
}
```